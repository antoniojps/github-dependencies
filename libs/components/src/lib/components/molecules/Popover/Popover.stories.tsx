import React, { useState } from 'react';
import { Popover } from './Popover';
import { User } from '../';

export default {
  title: 'molecules/Popover',
};

const Content = () => <div style={{ padding: '1rem' }}>Hello how are you?</div>;

export const Basic = () => {
  const [isOpen, setIsOpen] = useState(false);

  return (
    <Popover
      content={Content}
      isOpen={isOpen}
      onChangeOpen={setIsOpen}
      positions={['bottom']}
      padding={5}
    >
      <User image={'https://react.geist-ui.dev/images/avatar.png'} name={'Joe'} size={'large'} />
    </Popover>
  );
};
