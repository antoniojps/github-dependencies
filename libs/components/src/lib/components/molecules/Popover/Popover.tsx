import React, { useState, useEffect, ReactNode } from 'react';
import { Popover as TinyPopover, PopoverProps } from 'react-tiny-popover';
import styles from './Popover.module.scss';

type Props = {
  children: ReactNode;
  onChangeOpen?: (isOpen: boolean) => void;
} & PopoverProps;

export const Popover = ({ children, content, onChangeOpen, positions, align, ...rest }: Props) => {
  const [isOpen, setOpen] = useState(false);

  const handleClick = (e) => {
    e.preventDefault();
    setOpen(!isOpen);
  };

  useEffect(() => {
    onChangeOpen(isOpen);
  }, [isOpen]);

  const handleClickOutside = () => {
    setOpen(false);
  };

  return (
    <TinyPopover
      isOpen={isOpen}
      positions={positions}
      align={align}
      content={content}
      onClickOutside={handleClickOutside}
      containerStyle={{
        overflow: 'visible',
        backgroundColor: 'var(--foreground)',
        borderRadius: 'var(--radius)',
        color: 'var(--base-inverse)',
        zIndex: 'var(--z-index-m)',
      }}
      {...rest}
    >
      <div onClick={handleClick} className={styles.children}>
        {children}
      </div>
    </TinyPopover>
  );
};
