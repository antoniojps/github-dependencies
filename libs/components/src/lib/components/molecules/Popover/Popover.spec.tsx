import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';

import { Popover } from './Popover';

const Content = () => <div>content</div>;

const PopoverDummy = ({ isOpenDefault }: { isOpenDefault?: boolean }) => {
  const [isOpen, setIsOpen] = React.useState(isOpenDefault);

  return (
    <Popover
      content={Content}
      isOpen={isOpen}
      onChangeOpen={setIsOpen}
      positions={['bottom']}
      padding={5}
    >
      <p>click</p>
    </Popover>
  );
};

describe('Popover', () => {
  it('renders', () => {
    const { asFragment } = render(<PopoverDummy />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders open', () => {
    const { asFragment } = render(<PopoverDummy isOpenDefault />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('render popover content on click', () => {
    render(<PopoverDummy />);
    expect(screen.queryByText('content')).toBeNull();
    fireEvent.click(screen.getByText('click'));
    expect(screen.getByText('content')).toBeTruthy();
  });
});
