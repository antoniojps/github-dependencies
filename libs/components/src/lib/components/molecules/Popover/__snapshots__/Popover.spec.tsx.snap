// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover renders 1`] = `
<DocumentFragment>
  <div
    class="children"
  >
    <p>
      click
    </p>
  </div>
</DocumentFragment>
`;

exports[`Popover renders open 1`] = `
<DocumentFragment>
  <div
    class="children"
  >
    <p>
      click
    </p>
  </div>
</DocumentFragment>
`;
